assets/js/anchor_accordion.js

```diff
@@ -0,0 +1,16 @@
+(() => {
+  const onChange = (_event) => {
+    const urlHash = new URL(document.URL).hash;
+    const idFromHash = urlHash.match(/^(#[A-Za-z0-9_-]+)$/);
```

Comment:
Since this came up before I put the code up for review: I deliberately want to keep some input sanitization here. The format I've picked is in line with the general format we use for `class` and `id` attributes, so it feels appropriate to me.